pkg/virt-manifest/mapper.go

```diff
@@ -42,52 +43,40 @@ type savedDisk struct {
 	disk v1.Disk
 }
 
-func ExtractPvc(dom *v1.DomainSpec) (*v1.DomainSpec, []savedDisk) {
-	specCopy := &v1.DomainSpec{}
-	model.Copy(specCopy, dom)
-
+func ExtractPvc(dom *v1.DomainSpec) []savedDisk {
 	pvcDisks := []savedDisk{}
-	allDisks := []v1.Disk{}
 
-	for idx, disk := range specCopy.Devices.Disks {
+	for idx, disk := range dom.Devices.Disks {
 		if disk.Type == Type_PersistentVolumeClaim {
 			// Save the disk so we can fix it later
 			diskCopy := v1.Disk{}
 			model.Copy(&diskCopy, disk)
 			pvcDisks = append(pvcDisks, savedDisk{disk: diskCopy, idx: idx})
-
-			// Alter the disk record so that libvirt will accept it
-			disk.Type = Type_Network
-			disk.Source.Protocol = "iscsi"
 		}
-		allDisks = append(allDisks, disk)
 	}
-	// Replace the Domain's disks with modified records
-	specCopy.Devices.Disks = allDisks
 
-	return specCopy, pvcDisks
+	return pvcDisks
 }
 
 // This is a simplified version of the domain creation portion of SyncVM. This is intended primarily
 // for mapping the VM spec without starting a domain.
-func MapVM(con virtwrap.Connection, vm *v1.VM) (*v1.VM, error) {
+func MapVM(con virtwrap.Connection, vm *v1.VM, k8sClient kubernetes.Interface) (*v1.VM, error) {
```

Comment:
It will be the last admission controller when posting a VM (after possible others like Presets, ...). It is  the last one invoked when you post a VM, not when it is started.